docs/Catalog/Embed.md

```diff
@@ -0,0 +1,349 @@
+Command message data format:
+
+```ts
+interface Command {
+  type: string
+  // ...parameters
+}
+```
+
+Available commands are listed below.
+
+#### `init`
+
+Initialize the Embed.
+This command must be sent after the Embed is ready
+(see `ready` message reference for details).
+
+```ts
+interface InitCommand extends Command {
+  type: 'init'
+
+  // Bucket name, e.g. 'my-bucket'
+  bucket?: string
+  // Path to object or prefix in the given bucket, e.g. 'some-prefix/some-object.csv'
+  path?: string
+
+  // Initial route Embed will navigate to, e.g. '/b/my-bucket/tree/some/path',
+  // takes precedence over bucket / path
+  route?: string
+
+  // Embed accepts any credentials supported by the Quilt authentication endpoint,
+  // e.g. { provider, token } for SSO or { password, username } (which doesn't seem like a right choice in most cases tho).
+  // Getting credentials is your app's responsibility
+  credentials: { provider: string; token: string } | { username: string; password: string }
+
+  // Embed can be "scoped" to a prefix, meaning that prefix will be a virtual "root" for the object browser,
+  // but only for display purposes (i.e. formatting paths / rendering breadcrumbs),
+  // it won't prevent navigating to the paths outside the scope if navigated directly
+  // via a command (navigate or init) or a link, so it's not to be considered a security measure.
+  scope?: string
+
+  // Look and feel of the Embed can be customzied by providing theme overrides,
+  // see [MUI theming reference](https://material-ui.com/customization/theming/)
+  // and [Quilt theme construction code](https://github.com/quiltdata/quilt/blob/master/catalog/app/constants/style.js#L145)
+  // for details.
+  theme?: MUI.ThemeOptions // https://github.com/mui-org/material-ui/blob/v4.12.3/packages/material-ui/src/styles/createTheme.d.ts#L15
+
+  // Some aspects of the UI can be overriden:
+  overrides?: {
+    // This prop is responsible for customizing the display and behaviour
+    // of the "link" button in the object revision list menu
+    s3ObjectLink?: {
+      // Title of the link element
+      title?: string
+      // Link [template](https://lodash.com/docs/4.17.15#template).
+      // Template context:
+      //   url: string -- url / route of the object version in the context of the Embed
+      //   s3HttpsUri: string -- HTTPS URI of the object version, e.g. https://my-bucket.s3.amazonaws.com/${key}?versionId=${version} (with properly encoded key)
+      //   bucket: string -- current bucket
+      //   key: string -- key of the browsed object
+      //   version: string -- object version id
+      // Example: 'https://my-app/s3-browser?route=<%= encodeURIComponent(url) %>'
+      href?: string
+      // Notification shown after copying href to the clipboard (if `emit` is not set to "override")
+      notification?: string
+      // When set "notify" or "override", Embed will send "s3ObjectLink" message.
```

Comment:
```suggestion
      // When set "notify" or "override", Embed will send the "s3ObjectLink" message.
```
There should be alist of message types, even if it has only one item for now.

Reply:
other messages are not affected, it's only for `s3ObjectLink`